.claude/skills/cookbook-audit/validate_notebook.py

```diff
@@ -0,0 +1,386 @@
+
+Usage:
+    python validate_notebook.py <notebook.ipynb>
+
+Exit codes:
+    0 - No issues found
+    1 - Critical issues found (must fix)
+"""
+
+import json
+import sys
+import re
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import List, Tuple, Dict, Optional
+
+class NotebookValidator:
+    def __init__(self, notebook_path: str):
+        self.notebook_path = Path(notebook_path)
+        self.issues: List[str] = []
+        self.warnings: List[str] = []
+        self.markdown_output: Optional[Path] = None
+
+        if not self.notebook_path.exists():
+            raise FileNotFoundError(f"Notebook not found: {notebook_path}")
+
+        with open(self.notebook_path, 'r', encoding='utf-8') as f:
+            self.nb = json.load(f)
+
+        self.cells = self.nb.get('cells', [])
+
+    def get_cell_source(self, cell: Dict) -> str:
+        """Get cell source as a single string."""
+        source = cell.get('source', [])
+        if isinstance(source, list):
+            return ''.join(source)
+        return source
+
+    def convert_to_markdown(self) -> Path:
+        """
+        Convert notebook to markdown for easier review.
+        Returns path to the markdown file in a temp directory.
+        Includes code cells but excludes outputs to save context.
+        Uses uv to run jupyter nbconvert with dependencies.
+        """
+        try:
+            # Create temp directory within skill folder
+            skill_dir = Path(__file__).parent
+            temp_dir = skill_dir / "tmp"
+            temp_dir.mkdir(exist_ok=True)
+
+            # Generate output filename
+            output_file = temp_dir / f"{self.notebook_path.stem}_review.md"
+
+            # Use uv to run jupyter nbconvert with nbconvert dependency
+            # Include code but exclude outputs for cleaner review
+            cmd = [
+                "uv", "run", "--with", "nbconvert",
+                "jupyter", "nbconvert",
+                "--to", "markdown",
+                "--output", str(output_file.absolute()),
+                "--no-prompt",  # Remove input/output prompts
+                "--TemplateExporter.exclude_output=True",  # Exclude cell outputs
+                str(self.notebook_path.absolute())
+            ]
+
+            result = subprocess.run(
+                cmd,
+                capture_output=True,
+                text=True,
+                check=True
+            )
+
+            self.markdown_output = output_file
+            return output_file
+
+        except subprocess.CalledProcessError as e:
+            print(f"Warning: Could not convert to markdown: {e.stderr}")
+            return None
+        except FileNotFoundError:
+            print("Warning: uv not found. Install from https://github.com/astral-sh/uv")
+            return None
+
+    def check_hardcoded_secrets(self):
+        """Check for hardcoded API keys and secrets."""
+        patterns = {
```

Comment:
can we use something like [detect-secrets](https://github.com/Yelp/detect-secrets) for this?